pycbc/noise/reproduceable.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (C) 2017  Alex Nitz
+#
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+import numpy, pycbc.psd
+from pycbc.types import TimeSeries
+from numpy.random import RandomState
+
+# These need to be constant to be able to recover identical results.
+# The hope is that nobody needs a higher resolution
+SAMPLE_RATE = 16384
+BLOCK_SIZE = 100
+FILTER_LENGTH = 128
+
+def block(seed):
+    """ Return block of normal random numbers
+    
+    Parameters
+    ----------
+    seed : {None, int}
+        The seed to generate the noise.
+
+    Returns
+    --------
+    noise : numpy.ndarray
+        Array of random numbers
+    """ 
+    num = SAMPLE_RATE * BLOCK_SIZE
+    rng = RandomState(seed % 2**32)
+    variance = SAMPLE_RATE / 2
+    return rng.normal(size=num, scale=variance**0.5)
+
+def normal(start, end, seed=0):
+    """ Generate data with a white Gaussian (normal) distribution
+
+    Parameters
+    ----------
+    start_time : int
+        Start time in GPS seconds to generate noise
+    end_time : int
+        End time in GPS seconds to generate nosie
+    seed : {None, int}
+        The seed to generate the noise.
+
+    Returns
+    --------
+    noise : TimeSeries
+        A TimeSeries containing gaussian noise
+    """
+    # This is reproduceable because we used fixed seeds from known values
+    s = int(start / BLOCK_SIZE)
+    e = int(end / BLOCK_SIZE)
```

Comment:
In the case that end / BLOCK_SIZE = end // BLOCK_SIZE you'll get 100s of data extra.